jax/_src/random.py

```diff
@@ -230,7 +230,9 @@ def uniform(key: KeyArray,
                      f"got {dtype}")
   dtype = dtypes.canonicalize_dtype(dtype)
   shape = core.as_named_shape(shape)
-  return _uniform(key, shape, dtype, minval, maxval)  # type: ignore
+  out = _uniform(key, shape, dtype, minval, maxval)  # type: ignore
```

Comment:
Seeing all the use of `key.consume`, I am wondering if the code ends up more readable with a slightly different API where `key.consume()` is the identity function, and you write `return _uniform(key.consume(), shape, ...)`